tests/integration/targets/json_file/tasks/main.yml

```
@@ -0,0 +1,327 @@
+---
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# (c) 2022 DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: set dest_file
+  set_fact:
+    dest_file: "{{ remote_tmp_dir }}/file.json"
+
+- name: set test_case
+  set_fact:
+    test_cases:
+      - id: 010 fail when destination does not contain vaild json
+        current_file: has_no_valid_json.txt
+        expected_file: has_no_valid_json.txt
+        module_params:
+          state: "present"
+          value:
+            A: 1
+        assert:
+          - "last_result.failed == true"
+          - "last_result.changed == false"
+          - "last_result.msg == 'Failed to decode JSON in {{ dest_file }}'"
+      - id: 020 can work when create dest
+        current_file: false
+        expected_file: sample.json
+        module_params:
+          state: "present"
+          allow_creation: true
+          value:
+            B: 2
+            A: 1
+            C:
+              CB: 4
+              CA: 3
+              CC:
+                - 5
+                - 6
+                - 7
+        assert:
+          - "last_result.failed == false"
+          - "last_result.changed == true"
+          - "last_result.created == true"
```

```suggestion
          - last_result is not failed
          - last_result is changed
          - last_result.created == true
```
(Same in other places.)